internal/settings.go

```diff
@@ -165,31 +168,37 @@ func (ds *DialSettings) Validate() error {
 	return nil
 }
 
-// GetDefaultUniverseDomain returns the default service domain for a given Cloud
-// universe, as configured with internaloption.WithDefaultUniverseDomain.
-// The default value is "googleapis.com".
+// GetDefaultUniverseDomain returns the Google default universe domain
+// ("googleapis.com").
 func (ds *DialSettings) GetDefaultUniverseDomain() string {
-	if ds.DefaultUniverseDomain == "" {
-		return universeDomainDefault
-	}
-	return ds.DefaultUniverseDomain
+	return defaultUniverseDomain
 }
 
 // GetUniverseDomain returns the default service domain for a given Cloud
-// universe, as configured with option.WithUniverseDomain.
-// The default value is the value of GetDefaultUniverseDomain, as configured
-// with internaloption.WithDefaultUniverseDomain.
+// universe, with the following precedence:
+//
+// 1. A non-empty option.WithUniverseDomain.
+// 2. A non-empty environment variable GOOGLE_CLOUD_UNIVERSE_DOMAIN.
+// 3. The default value "googleapis.com".
 func (ds *DialSettings) GetUniverseDomain() string {
-	if ds.UniverseDomain == "" {
-		return ds.GetDefaultUniverseDomain()
+	ds.udMu.Lock()
```

Comment:
Instead of updating this value and adding a mutex can we just return the values?

Reply:
Yes, that should be fine with the new `auth` library, which will only access `GetUniverseDomain` on new client initialization afaik.

Reply:
Updated, thanks.